geomstats/geometry/positive_reals.py

```diff
@@ -0,0 +1,315 @@
+        base_point : array-like, shape=[..., 1]
+            Base point.
+
+        Returns
+        -------
+        inner_product : array-like, shape=[...,]
+            Inner-product.
+        """
+        inner_product_matrix = self.inner_product_matrix(base_point=base_point)
+        inner_product = tangent_vec_a * inner_product_matrix * tangent_vec_b
+        inner_product = gs.reshape(inner_product, (-1,))
+        return inner_product
+
+    def squared_norm(self, vector, base_point):
+        """Compute the square of the norm of a vector.
+
+        Squared norm of a vector associated to the inner product
+        at the tangent space at a base point.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., 1]
+            Vector.
+        base_point : array-like, shape=[..., 1]
+            Base point.
+            Optional, default: None.
+
+        Returns
+        -------
+        sq_norm : array-like, shape=[...,]
+            Squared norm.
+        """
+        sq_norm = self.inner_product(vector, vector, base_point=base_point)
+        return gs.real(sq_norm)
+
+    @staticmethod
+    def exp(tangent_vec, base_point):
+        """Compute the positive reals exponential map.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., 1]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., 1]
+            Base point.
+
+        Returns
+        -------
+        exp : array-like, shape=[..., 1]
+            Riemannian exponential.
+        """
+        exp = base_point * gs.exp(tangent_vec / base_point)
+        return exp
+
+    @staticmethod
+    def log(point, base_point):
+        """Compute the positive reals logarithm map.
+
+        Compute the Riemannian logarithm at point base_point,
+        of point wrt the positive reals metric.
+        This gives a tangent vector at point base_point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point.
+        base_point : array-like, shape=[..., 1]
+            Base point.
+
+        Returns
+        -------
+        log : array-like, shape=[..., 1]
+            Riemannian logarithm.
+        """
+        log = base_point * gs.log(point / base_point)
+        return log
+
+    def squared_dist(self, point_a, point_b):
+        """Compute the positive reals squared distance.
+
+        Compute the Riemannian squared distance between point_a and point_b.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., 1]
+            Point.
+        point_b : array-like, shape=[..., 1]
+            Point.
+
+        Returns
+        -------
+        squared_dist : array-like, shape=[...,]
+            Riemannian squared distance.
+        """
+        sq_dist = gs.log(point_b / point_a) ** 2
+        sq_dist = gs.reshape(sq_dist, (-1,))
+        sq_dist *= self.scale**2
+        return gs.real(sq_dist)
```

Comment:
again, shouldn't it be real already?

Reply:
Once `PositiveReals` and `ComplexPoincareDisk` are both merged, I will create a product manifold named `ProductPositiveRealsAndComplexPoincareDisks`. 
In this product manifold, the default dtype is complex.
Therefore, it is very convenient if `PositiveReals` can accept both reals a complex data types.